test/unit/geo/transform.test.js

```diff
@@ -117,6 +117,112 @@ test('transform', (t) => {
         t.end();
     });
 
+    t.test('maxBounds should not jump to the wrong side when crossing 180th meridian (#10447)', (t) => {
+        t.test(' to the East', (t) => {
+            const transform = new Transform();
+            transform.zoom = 6;
+            transform.resize(500, 500);
+            transform.lngRange = [160, 190];
+            transform.latRange = [-55, -23];
+
+            transform.center = new LngLat(-170, -40);
```

Comment:
Yes, this test sets the center to the east bound in order to test that the map position snaps to the correct side.  The exact position is arbitrary, it could be any position out of bounds on the east side of the map.